app/ui/components/Button/NavButton.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+import { Routes } from "@/app/routes/routes";
+import Button from "@/app/ui/components/Button/Button";
+import { ButtonType } from "@/app/ui/components/Button/utils";
+import { signIn } from "next-auth/react";
+
+export const NavButton = ({
```

Comment:
nav button is not really good component (designing such way you will have hundreds button componenets soon). Move such logic to bigger or "clever" component (for example you can keep handleSignIn function in separate file and reuse it if in future users will be able to sign in from different button\links\actions\events)